More readable colors in 'git hist'

jan-warchol opened the pull request:
before and after image:
![git-hist-before-and-after](https://user-images.githubusercontent.com/1368508/48184221-a242f580-e331-11e8-82d3-1d18500162eb.png)

* use "auto" (yellow) instead of red for commit hashes
  Having so much red in the output sends an alarming message to the
  viewer, as red is associated with errors.

* use "auto" instead of yellow for ref names
  This makes git color refs according to their type: tags yellow, HEAD
  cyan, local branches green and remote branches red. Different colors
  make navigating through the log and finding desired branch/tag easier.
  Also, use blue (instead of default red) for remote branches.

* put ref names at the beginning of line
  Sometimes there are multiple branches and/or tags pointing to a ref,
  and the list may be so long that the first line won't fit on the
  screen. Since branch information is more important than authorship,
  put it first so that more of it is visible in such cases

* use "white dim" (grey) for author and date
  This is secondary content, and it repeats a lot across commits, so it
  shouldn't be highlighted in bold and intense colors. Rather, it should
  be dimmer than commit message, to make searching for a particular
  commit (by its message) or branch (by its name) easier.

jakubnabrdalik commented:
Hi @jan-warchol 

Sorry for the extremaly long wait of this PR. I have no idea how have I missed it before (even though I completely suck as a maintaner of this repo in general).

Back to the  topic

> use "auto" (yellow) instead of red for commit hashes

I agree with you. While I like the difference in color, red has also the problem of being hard to read if you monitor has low brightness (it's a pretty dark red). On OLED displays it does look distinctive, but most screens are LCDs.

> use "auto" instead of yellow for ref names

jan-warchol commented:
Hi @jakubnabrdalik 

> Sorry for the extremaly long wait of this PR. I have no idea how have I missed it before (even though I completely suck as a maintaner of this repo in general).

It took me ages to reply, too :p

> > use "white dim" (grey) for author and date
> 
> That one I personally do not like. The dim (grey) color is hard to see on some LCDs (or in high resolution, when you are an old person, like me), and both author and date are very important usually.

It also depends a lot on the terminal's color palette. Which makes me wonder what palette you are using :-) You may be interested in taking a look at https://github.com/jan-warchol/selenized, it's a science-based color palette for terminals with carefully tuned readability. It got a decent number of stars already.
 
> Would you be ok if I took half of your PR (first two points) only?
> Would you like me to prepare this change, or would you like to do it yourself (so that the commit has your name on it)?

Sure. I rewrote the commit (updating `NSFW` configs as well). Since I have deleted my for in the meantime, it seems I have to open a new PR for the updated version.

jan-warchol commented:
Since I have deleted my for in the meantime, it seems I have to open a new PR for the updated version. See #68
